press/press/doctype/deploy_candidate/utils.py

```diff
@@ -142,16 +145,31 @@ def get_build_server(group: str | None = None) -> str | None:
 	if group and (server := frappe.get_value("Release Group", group, "build_server")):
 		return server
 
-	if server := get_build_server_with_least_active_builds():
+	if group:
+		release_group: ReleaseGroup = frappe.get_doc("Release Group", group)
+		for server in release_group.servers:
+			server_platform = frappe.get_value("Server", server.server, "platform")
+			if server_platform == "arm64" and (server := get_arm_build_server_with_least_active_builds()):
+				return server
+
+	if server := get_x86_build_server_with_least_active_builds():
 		return server
```

Comment:
There might be one issue.

Assume, you have a public release group and you have multiple servers linked with it (some are from AWS, some from OCI). 

And For now you can only move aws servers to ARM, not the OCI ones.

So, in that case you need to build for both platform ?

Reply:
That's an issue as well, I was thinking how do we handle this but for now I decided to reduce complexity maybe?

Reply:
Yeah mostly would be `public` servers for shared hosting. Those can be ignored then in first phase of migration.

Reply:
Maybe only release groups with one server should be targeted in the first phase then 🤔